### PR TITLE
Enhancement and Bugfix in 2.7.1 ( Part 3 ) 

### PR DESCRIPTION
**Pull request by mercyblitz**
## Enhancement

- service register support on nacos: #3582 

## Bugfix

- `@Reference` bean name conflict #1306

### CODE REVIEW

File: dubbo-registry/dubbo-registry-nacos/pom.xml [hunk 1]
```diff
@@ -0,0 +1,146 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (beiwei30) [inline dubbo-registry/dubbo-registry-nacos/pom.xml]: copyright header.

File: dubbo-registry/dubbo-registry-nacos/pom.xml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
```
Comment (beiwei30) [inline dubbo-registry/dubbo-registry-nacos/pom.xml]: hard coded **4.12**

File: dubbo-registry/dubbo-registry-nacos/pom.xml [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            <groupId>org.apache.dubbo</groupId>
+            <artifactId>dubbo-remoting-netty4</artifactId>
+            <version>${project.version}</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>ch.qos.logback</groupId>
+            <artifactId>logback-classic</artifactId>
+            <version>1.2.3</version>
```
Comment (beiwei30) [inline dubbo-registry/dubbo-registry-nacos/pom.xml]: hard coded **1.2.3**

File: dubbo-registry/dubbo-registry-nacos/pom.xml (hunk 2 above)
Comment (mercyblitz) [reply dubbo-registry/dubbo-registry-nacos/pom.xml]: I removed the junit, will use  jupiter as default.

File: dubbo-registry/dubbo-registry-nacos/src/test/resources/META-INF/spring/dubbo-consumer-context.xml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+       xmlns:dubbo="http://dubbo.apache.org/schema/dubbo"
+       xsi:schemaLocation="http://www.springframework.org/schema/beans        http://www.springframework.org/schema/beans/spring-beans-4.3.xsd        http://dubbo.apache.org/schema/dubbo        http://dubbo.apache.org/schema/dubbo/dubbo.xsd">
+
+    <!-- 提供方应用信息，用于计算依赖关系 -->
+    <dubbo:application name="dubbo-consumer-xml-demo"/>
+
+    <!-- 使用 Nacos 注册中心 -->
+    <dubbo:registry address="nacos://127.0.0.1:8848"/>
+
+    <!-- 引用服务接口 -->
```
Comment (lovepoem) [inline dubbo-registry/dubbo-registry-nacos/src/test/resources/META-INF/spring/dubbo-consumer-context.xml]: Please translate the comment

**Comment by lovepoem** [comment]
lgtm

**Pull request merged**